In [1]:
import pandas as pd

In [2]:
data=pd.read_csv('Ecommerce Purchases')
data

Address    Lot AM or PM  \
0     16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                      Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3                 7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4     23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
...                                                 ...    ...      ...   
9995      966 Castaneda Locks\nWest Juliafurt, CO 96415  92 XI       PM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
0     Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2     Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3     Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4     Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
...                                                 ...   
9995  Mozilla/5.0 (Windows NT 5.1) AppleWebKit/5352 ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                              Company       Credit Card CC Exp Date  \
0                     Martinez-Herman  6011929061123406       02/20   
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
2          Simpson, Williams and Pham      675957666125       08/19   
3     Williams, Marshall and Buchanan  6011578504430710       02/24   
4           Brown, Watson and Andrews  6011456623207998       10/25   
...                               ...               ...         ...   
9995                    Randall-Sloan   342945015358701       03/22   
9996         Hale, Collins and Wilson   210033169205009       07/25   
9997                     Anderson Ltd  6011539787356311       05/21   
9998                         Cook Inc   180003348082930       11/17   
9999                       Greene Inc  4139972901927273       02/19   

      CC Security Code                  CC Provider  \
0                  900                 JCB 16 digit   
1                  561                   Mastercard   
2                  699                 JCB 16 digit   
3                  384                     Discover   
4                  678  Diners Club / Carte Blanche   
...                ...                          ...   
9995               838                 JCB 15 digit   
9996               207                 JCB 16 digit   
9997                 1                VISA 16 digit   
9998               987             American Express   
9999               302                 JCB 15 digit   

                               Email                                     Job  \
0                  pdunlap@yahoo.com  Scientist, product/process development   
1                 anthony41@reed.com                       Drilling engineer   
2     amymiller@morales-harrison.com                Customer service manager   
3        brent16@olson-robinson.info                       Drilling engineer   
4        christopherwright@gmail.com                             Fine artist   
...                              ...                                     ...   
9995          iscott@wade-garner.com                              Printmaker   
9996              mary85@hotmail.com                         Energy engineer   
9997               tyler16@gmail.com                      Veterinary surgeon   
9998         elizabethmoore@reid.net                Local government officer   
9999           rachelford@vaughn.com        

Display top 10 rows of dataset

In [3]:
data.head(10)

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
5  7502 Powell Mission Apt. 768\nTravisland, VA 3...  21 XT       PM   
6     93971 Conway Causeway\nAndersonburgh, AZ 75107  96 Xt       AM   
7  260 Rachel Plains Suite 366\nCastroberg, WV 24...  96 pG       PM   
8            2129 Dylan Burg\nNew Michelle, ME 28650  45 JN       PM   
9    3795 Dawson Extensions\nLake Tinafort, ID 88739  15 Ug       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
5  Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10_8_5...   
6  Mozilla/5.0 (compatible; MSIE 7.0; Windows NT ...   
7  Mozilla/5.0 (X11; Linux i686) AppleWebKit/5350...   
8  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9  Mozilla/5.0 (X11; Linux i686; rv:1.9.7.20) Gec...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   
4        Brown, Watson and Andrews  6011456623207998       10/25   
5                   Silva-Anderson    30246185196287       07/25   
6                  Gibson and Sons  6011398782655569       07/24   
7                 Marshall-Collins      561252141909       06/25   
8                Galloway and Sons   180041795790001       04/24   
9     Rivera, Buchanan and Ramirez     4396283918371       01/17   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   
5              7169                     Discover   
6               714                VISA 16 digit   
7               256                VISA 13 digit   
8               899                 JCB 16 digit   
9               931             American Express   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   
5               ynguyen@gmail.com                       Fish farm manager   
6              olivia04@yahoo.com                                  Dancer   
7            phillip48@parks.info                         Event organiser   
8            kdavis@rasmussen.com                       Financial manager   
9        qcoleman@hunt-huerta.com                      Forensic scientist   

        IP Address Language  Purchase Price  
0  149.146.147.205       el           98.14  
1     15.160.41.51       fr           70.73  
2   132.207.160.22       de            0.95  
3     30.250.74.19       es           78.04  
4     24.140.33.94       es           77.82  
5    55.96.152.147       ru           25.15  
6   127.252.144.18       de           88.56  
7   224.247.97.150       pt           44.25  
8  146.234.201.229       ru           59.54  
9    

Display last 10 rows of dataset

In [4]:
data.tail(10)

Address    Lot AM or PM  \
9990  75731 Molly Springs\nWest Danielle, VT 96934-5102  93 ty       PM   
9991              PSC 8165, Box 8498\nAPO AP 60327-0346  50 dA       AM   
9992  885 Allen Mountains Apt. 230\nWallhaven, LA 16995  40 vH       PM   
9993  7555 Larson Locks Suite 229\nEllisburgh, MA 34...  72 jg       PM   
9994       6276 Rojas Hollow\nLake Louis, WY 56410-7837  93 Ex       PM   
9995      966 Castaneda Locks\nWest Juliafurt, CO 96415  92 XI       PM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
9990  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_7_4;...   
9991  Mozilla/5.0 (compatible; MSIE 8.0; Windows NT ...   
9992  Mozilla/5.0 (Macintosh; PPC Mac OS X 10_6_5) A...   
9993  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_8...   
9994  Opera/9.68.(X11; Linux x86_64; sl-SI) Presto/2...   
9995  Mozilla/5.0 (Windows NT 5.1) AppleWebKit/5352 ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                         Company       Credit Card CC Exp Date  \
9990  Pace, Vazquez and Richards   869968197049750       04/24   
9991                  Snyder Inc  4221582137197481       02/24   
9992                   Wells Ltd  4664825258997302       10/20   
9993              Colon and Sons    30025560104631       10/25   
9994                Ritter-Smith  3112186784121077       01/25   
9995               Randall-Sloan   342945015358701       03/22   
9996    Hale, Collins and Wilson   210033169205009       07/25   
9997                Anderson Ltd  6011539787356311       05/21   
9998                    Cook Inc   180003348082930       11/17   
9999                  Greene Inc  4139972901927273       02/19   

      CC Security Code       CC Provider                        Email  \
9990               877      JCB 15 digit  andersonmichael@sherman.biz   
9991               969           Voyager           kking@wise-liu.com   
9992               431          Discover            bberry@wright.net   
9993               629           Maestro    chelseawilliams@lopez.biz   
9994              1823           Maestro           iroberts@gmail.com   
9995               838      JCB 15 digit       iscott@wade-garner.com   
9996               207      JCB 16 digit           mary85@hotmail.com   
9997                 1     VISA 16 digit            tyler16@gmail.com   
9998               987  American Express      elizabethmoore@reid.net   
9999               302      JCB 15 digit        rachelford@vaughn.com   

                               Job       IP Address Language  Purchase Price  
9990           Early years teacher     54.170.3.185       ru           18.35  
9991         IT sales professional    254.25.31.156       el           25.93  
9992                  Set designer    174.173.51.32       de           67.96  
9993  Designer, exhibition/display    177.46.82.128       el           65.61  
9994     Education officer, museum    242.44.112.18       zh           31.85  
9995                    Printmaker    29.73.197.114       it           82.21  
9996               Energy engineer   121.133.168.51       pt           25.63  
9997            Veterinary surgeon    156.210.0.254       el           83.98  
9998      Local government officer     55.78.26.143       es           38.84  
9999        Embryologist, clinical  176.119.198.199       el           67.59

Check datatype of each column

In [5]:
data.dtypes

Address              object
Lot                  object
AM or PM             object
Browser Info         object
Company              object
Credit Card           int64
CC Exp Date          object
CC Security Code      int64
CC Provider          object
Email                object
Job                  object
IP Address           object
Language             object
Purchase Price      float64
dtype: object

Checck for null (NA) values in dataset


In [6]:
data.isnull().sum()

Address             0
Lot                 0
AM or PM            0
Browser Info        0
Company             0
Credit Card         0
CC Exp Date         0
CC Security Code    0
CC Provider         0
Email               0
Job                 0
IP Address          0
Language            0
Purchase Price      0
dtype: int64

What does our dataset look like

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


Find highest,lowest and average purchase price

In [8]:
data['Purchase Price']

0       98.14
1       70.73
2        0.95
3       78.04
4       77.82
        ...  
9995    82.21
9996    25.63
9997    83.98
9998    38.84
9999    67.59
Name: Purchase Price, Length: 10000, dtype: float64

In [9]:
data['Purchase Price'].max()
#print('Highest purchase price is ',data['Purchase Price'].max())

99.99

In [10]:
data['Purchase Price'].min()
#print('Lowest purchase price is ',data['Purchase Price'].min())

0.0

In [11]:
data['Purchase Price'].mean()
#print('Average purchase price is ',data['Purchase Price'].mean())


50.34730200000025

How many people have French as their language?

In [12]:
data.Language.value_counts()

de    1155
ru    1155
el    1137
pt    1118
en    1098
fr    1097
es    1095
it    1086
zh    1059
Name: Language, dtype: int64

Hence 1097 people have French (Fr) as their language.


In [13]:
#another way to creach this result
french=data[data['Language']=='fr']  #a slice of the data that includes only french speaking ppl

In [14]:
len(french)

1097

Find job title contains 'Engineer'

In [15]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [16]:
data.Job

0       Scientist, product/process development
1                            Drilling engineer
2                     Customer service manager
3                            Drilling engineer
4                                  Fine artist
                         ...                  
9995                                Printmaker
9996                           Energy engineer
9997                        Veterinary surgeon
9998                  Local government officer
9999                    Embryologist, clinical
Name: Job, Length: 10000, dtype: object

In [17]:
data.Job.dtypes

dtype('O')

Its an object type so we need the string function

In [18]:
data[data.Job.str.contains('Engineer',case=False)]  #case=false makes the search case insensitive

Address    Lot AM or PM  \
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
3                 7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
50      41159 Michael Centers\nAdamsfort, RI 37108-6674  46 Ce       PM   
55    27635 Maureen Bypass Apt. 883\nSandraview, SD ...  59 LJ       AM   
60    7126 Katherine Squares\nPerkinsview, CO 97299-...  63 qu       AM   
...                                                 ...    ...      ...   
9948  95544 Johnson Isle Suite 939\nMichaelberg, RI ...  91 bW       AM   
9952            9991 Vaughn Hills\nRacheltown, PA 55409  36 KC       PM   
9970              0060 Keith Stream\nWestport, CO 47097  11 nt       PM   
9977  02182 Keith Expressway\nEast Shannon, CT 20578...  34 RL       AM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   

                                           Browser Info  \
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
3     Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
50    Mozilla/5.0 (Windows 98; Win 9x 4.90; sl-SI; r...   
55    Mozilla/5.0 (iPod; U; CPU iPhone OS 3_3 like M...   
60    Opera/8.68.(X11; Linux x86_64; en-US) Presto/2...   
...                                                 ...   
9948  Opera/8.36.(X11; Linux x86_64; sl-SI) Presto/2...   
9952  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   
9970  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_8...   
9977  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   

                              Company       Credit Card CC Exp Date  \
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
3     Williams, Marshall and Buchanan  6011578504430710       02/24   
50        Wright, Williams and Mendez  4008586485908075       05/19   
55                         Sims-Lyons  3158113629128344       09/19   
60                 Marshall-Fernandez   349767747049645       05/20   
...                               ...               ...         ...   
9948                     Fox-Peterson     4762924304307       03/17   
9952         Ward, Smith and Castillo  6011679271321726       09/19   
9970         Carpenter, Good and Hart  6011485664704662       07/19   
9977      Deleon, Jacobson and Benton  4186094003664688       06/21   
9996         Hale, Collins and Wilson   210033169205009       07/25   

      CC Security Code    CC Provider                        Email  \
1                  561     Mastercard           anthony41@reed.com   
3                  384       Discover  brent16@olson-robinson.info   
50                 945   JCB 16 digit    susanvalentine@obrien.org   
55                 857  VISA 16 digit       adkinsarthur@yahoo.com   
60                 672   JCB 15 digit      sweeneyhannah@jones.biz   
...                ...            ...                          ...   
9948               567     Mastercard      haleybenjamin@gmail.com   
9952               964        Voyager      jonesjennifer@olson.com   
9970               543       Discover      rangelbrian@hotmail.com   
9977               397   JCB 16 digit       daltoncarter@yahoo.com   
9996               207   JCB 16 digit           mary85@hotmail.com   

                                      Job       IP Address Language  \
1                       Drilling engineer     15.160.41.51       fr   
3                       Drilling engineer     30.250.74.19       es   
50                    Mechanical engineer  213.203.143.215       de   
55    Engineer, broadcasting (operations)   227.235.89.210       pt   
60                 Engineer, agricultural  197.144.142.102       de   
...                                   ...              ...      ...   
9948                  Structural engineer    120.36.140.58       en   
9952                     Engineer, energy    116.228.12.42       es   
9970                  Electrical engineer     242.8.85.205       en   
9977                  Biomedical e

FInd details of the person with IP add: 132.207.160.22


In [19]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [20]:
data[data['IP Address']=='132.207.160.22']

Address    Lot AM or PM  \
2  Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   

                                        Browser Info  \
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   

                      Company   Credit Card CC Exp Date  CC Security Code  \
2  Simpson, Williams and Pham  675957666125       08/19               699   

    CC Provider                           Email                       Job  \
2  JCB 16 digit  amymiller@morales-harrison.com  Customer service manager   

       IP Address Language  Purchase Price  
2  132.207.160.22       de            0.95

How many people have mastercard as their CC provider

In [21]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [22]:
master_card=data[data['CC Provider'].str.contains('MasterCard',case=False)]

How many of these people have made purchase more than 50?

In [23]:
master_card[master_card['Purchase Price']>50]
#master_card[master_card['Purchase Price']>50].count()


Address    Lot AM or PM  \
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
18    461 Christopher Square\nWest Michaelchester, C...  17 SB       PM   
31                     USNS Alvarado\nFPO AA 27052-1231  26 Lh       PM   
35    93392 Webb Gardens Apt. 220\nLaurabury, AR 999...  37 om       AM   
90          431 Bowen Lights\nFergusonborough, MH 01362  31 tG       AM   
...                                                 ...    ...      ...   
9941           43757 Brown Lodge\nValerieberg, PR 88518  68 DG       AM   
9948  95544 Johnson Isle Suite 939\nMichaelberg, RI ...  91 bW       AM   
9954  051 Samantha Wells Apt. 328\nScottborough, OR ...  11 vG       AM   
9981  087 Larson Roads Apt. 587\nRicestad, WI 98077-...  95 Kw       AM   
9987     2754 Klein Mission\nJohnsonview, PA 64925-9624  65 Fg       PM   

                                           Browser Info  \
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
18    Mozilla/5.0 (X11; Linux i686; rv:1.9.6.20) Gec...   
31    Opera/8.84.(X11; Linux i686; sl-SI) Presto/2.9...   
35    Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
90    Mozilla/5.0 (Windows NT 5.2; it-IT; rv:1.9.1.2...   
...                                                 ...   
9941  Mozilla/5.0 (compatible; MSIE 5.0; Windows NT ...   
9948  Opera/8.36.(X11; Linux x86_64; sl-SI) Presto/2...   
9954  Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10_8_2...   
9981  Opera/9.63.(Windows NT 5.01; en-US) Presto/2.9...   
9987  Opera/9.26.(Windows CE; sl-SI) Presto/2.9.188 ...   

                              Company       Credit Card CC Exp Date  \
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
18             Beard, Abbott and Pena  6011350184276270       12/22   
31                    Nicholson Group     4614997834548       03/22   
35                           Mora Ltd  6011049630969815       09/16   
90                       Copeland-Lee      639032576097       06/18   
...                               ...               ...         ...   
9941         Jones, Williams and Dunn  6011508474487291       04/26   
9948                     Fox-Peterson     4762924304307       03/17   
9954    Mendoza, Zimmerman and Reilly    30232295374907       05/21   
9981         Ortiz, Schultz and Adams     4690863684605       01/19   
9987                       Curtis LLC   180027305702263       12/24   

      CC Security Code CC Provider                       Email  \
1                  561  Mastercard          anthony41@reed.com   
18                 767  Mastercard          hannah63@yahoo.com   
31                 909  Mastercard        ashley12@hotmail.com   
35                 367  Mastercard      hgonzalez@mcdowell.com   
90                 868  Mastercard           nrogers@brown.com   
...                ...         ...                         ...   
9941               906  Mastercard       christian55@gmail.com   
9948               567  Mastercard     haleybenjamin@gmail.com   
9954               762  Mastercard        brownamy@perkins.com   
9981               695  Mastercard  laurenbennett@richards.com   
9987               865  Mastercard    denisehamilton@novak.biz   

                                         Job       IP Address Language  \
1                          Drilling engineer     15.160.41.51       fr   
18                              Photographer   73.250.176.201       el   
31                           Sales executive   94.176.142.201       en   
35               Lecturer, further education     216.23.95.40       zh   
90    Teacher, English as a foreign language   153.188.13.203       es   
...                                      ...              ...      ...   
9941                Secondary school teacher  216.118.235.145       fr   
9948                     Structural engineer    120.36.140.58       en   
9954                 Presenter, broadcasting     61.97.84.101       it   
9981         Environmental education officer    106.

In [24]:
master_card[master_card['Purchase Price']>50].count()

Address             405
Lot                 405
AM or PM            405
Browser Info        405
Company             405
Credit Card         405
CC Exp Date         405
CC Security Code    405
CC Provider         405
Email               405
Job                 405
IP Address          405
Language            405
Purchase Price      405
dtype: int64

So 405 people fall in this category(MasterCard and purchases above 50)
Another way to do this

In [25]:
data[(data['CC Provider']=='Mastercard') & (data['Purchase Price']>50)]

Address    Lot AM or PM  \
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
18    461 Christopher Square\nWest Michaelchester, C...  17 SB       PM   
31                     USNS Alvarado\nFPO AA 27052-1231  26 Lh       PM   
35    93392 Webb Gardens Apt. 220\nLaurabury, AR 999...  37 om       AM   
90          431 Bowen Lights\nFergusonborough, MH 01362  31 tG       AM   
...                                                 ...    ...      ...   
9941           43757 Brown Lodge\nValerieberg, PR 88518  68 DG       AM   
9948  95544 Johnson Isle Suite 939\nMichaelberg, RI ...  91 bW       AM   
9954  051 Samantha Wells Apt. 328\nScottborough, OR ...  11 vG       AM   
9981  087 Larson Roads Apt. 587\nRicestad, WI 98077-...  95 Kw       AM   
9987     2754 Klein Mission\nJohnsonview, PA 64925-9624  65 Fg       PM   

                                           Browser Info  \
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
18    Mozilla/5.0 (X11; Linux i686; rv:1.9.6.20) Gec...   
31    Opera/8.84.(X11; Linux i686; sl-SI) Presto/2.9...   
35    Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
90    Mozilla/5.0 (Windows NT 5.2; it-IT; rv:1.9.1.2...   
...                                                 ...   
9941  Mozilla/5.0 (compatible; MSIE 5.0; Windows NT ...   
9948  Opera/8.36.(X11; Linux x86_64; sl-SI) Presto/2...   
9954  Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10_8_2...   
9981  Opera/9.63.(Windows NT 5.01; en-US) Presto/2.9...   
9987  Opera/9.26.(Windows CE; sl-SI) Presto/2.9.188 ...   

                              Company       Credit Card CC Exp Date  \
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
18             Beard, Abbott and Pena  6011350184276270       12/22   
31                    Nicholson Group     4614997834548       03/22   
35                           Mora Ltd  6011049630969815       09/16   
90                       Copeland-Lee      639032576097       06/18   
...                               ...               ...         ...   
9941         Jones, Williams and Dunn  6011508474487291       04/26   
9948                     Fox-Peterson     4762924304307       03/17   
9954    Mendoza, Zimmerman and Reilly    30232295374907       05/21   
9981         Ortiz, Schultz and Adams     4690863684605       01/19   
9987                       Curtis LLC   180027305702263       12/24   

      CC Security Code CC Provider                       Email  \
1                  561  Mastercard          anthony41@reed.com   
18                 767  Mastercard          hannah63@yahoo.com   
31                 909  Mastercard        ashley12@hotmail.com   
35                 367  Mastercard      hgonzalez@mcdowell.com   
90                 868  Mastercard           nrogers@brown.com   
...                ...         ...                         ...   
9941               906  Mastercard       christian55@gmail.com   
9948               567  Mastercard     haleybenjamin@gmail.com   
9954               762  Mastercard        brownamy@perkins.com   
9981               695  Mastercard  laurenbennett@richards.com   
9987               865  Mastercard    denisehamilton@novak.biz   

                                         Job       IP Address Language  \
1                          Drilling engineer     15.160.41.51       fr   
18                              Photographer   73.250.176.201       el   
31                           Sales executive   94.176.142.201       en   
35               Lecturer, further education     216.23.95.40       zh   
90    Teacher, English as a foreign language   153.188.13.203       es   
...                                      ...              ...      ...   
9941                Secondary school teacher  216.118.235.145       fr   
9948                     Structural engineer    120.36.140.58       en   
9954                 Presenter, broadcasting     61.97.84.101       it   
9981         Environmental education officer    106.

In [26]:
len(data[(data['CC Provider']=='Mastercard') & (data['Purchase Price']>50)])

405

We reach the same conclusion. 405 who use Mastercard have made purchases above 50!!

Find email of the person with the following credit card number:
4690863684605

In [27]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [28]:
data[data['Credit Card']==4690863684605]['Email']

9981    laurenbennett@richards.com
Name: Email, dtype: object

Now lets analyse,how many people shopped during the daytime(AM), and how ,any during the later half of the day (PM)?

In [29]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [43]:
data.groupby('AM or PM')

In [89]:
list(data.groupby('AM or PM'))
AM=list(data.groupby('AM or PM'))[0][1]
AM

Address    Lot AM or PM  \
4     23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
6        93971 Conway Causeway\nAndersonburgh, AZ 75107  96 Xt       AM   
9       3795 Dawson Extensions\nLake Tinafort, ID 88739  15 Ug       AM   
12      733 Heather Rest Apt. 670\nBoltonport, UT 78662  69 DO       AM   
14    8774 Jason Keys Suite 427\nEast Scottborough, ...  70 zH       AM   
...                                                 ...    ...      ...   
9989         5674 Cruz Trace\nNew Shelby, KY 51047-5469  12 Kf       AM   
9991              PSC 8165, Box 8498\nAPO AP 60327-0346  50 dA       AM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
4     Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
6     Mozilla/5.0 (compatible; MSIE 7.0; Windows NT ...   
9     Mozilla/5.0 (X11; Linux i686; rv:1.9.7.20) Gec...   
12    Mozilla/5.0 (Macintosh; Intel Mac OS X 10_5_3 ...   
14    Mozilla/5.0 (Windows 98; it-IT; rv:1.9.2.20) G...   
...                                                 ...   
9989  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9991  Mozilla/5.0 (compatible; MSIE 8.0; Windows NT ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                           Company       Credit Card CC Exp Date  \
4        Brown, Watson and Andrews  6011456623207998       10/25   
6                  Gibson and Sons  6011398782655569       07/24   
9     Rivera, Buchanan and Ramirez     4396283918371       01/17   
12                    Moore-Martin  5115990487067905       05/26   
14        Leach, Howe and Ferguson   869967499275071       09/22   
...                            ...               ...         ...   
9989  Vargas, Nichols and Martinez  5278049688438382       12/23   
9991                    Snyder Inc  4221582137197481       02/24   
9996      Hale, Collins and Wilson   210033169205009       07/25   
9997                  Anderson Ltd  6011539787356311       05/21   
9999                    Greene Inc  4139972901927273       02/19   

      CC Security Code                  CC Provider  \
4                  678  Diners Club / Carte Blanche   
6                  714                VISA 16 digit   
9                  931             American Express   
12                 119                VISA 16 digit   
14                 427                VISA 16 digit   
...                ...                          ...   
9989               996                VISA 16 digit   
9991               969                      Voyager   
9996               207                 JCB 16 digit   
9997                 1                VISA 16 digit   
9999               302                 JCB 15 digit   

                            Email                           Job  \
4     christopherwright@gmail.com                   Fine artist   
6              olivia04@yahoo.com                        Dancer   
9        qcoleman@hunt-huerta.com            Forensic scientist   
12              tholt@hotmail.com            Surveyor, quantity   
14               aburns@yahoo.com                 Acupuncturist   
...                           ...                           ...   
9989   walterswilliam@hotmail.com  Community development worker   
9991           kking@wise-liu.com         IT sales professional   
9996           mary85@hotmail.com               Energy engineer   
9997            tyler16@gmail.com            Veterinary surgeon   
9999        rachelford@vaughn.com        Embryologist, clinical   

           IP Address Language  Purchase Price  
4        24.140.33.94       es           77.82  
6      127.252.144.18       de           88.56  
9       

In [35]:
list(data.groupby('AM or PM'))[0][1].count()

Address             4932
Lot                 4932
AM or PM            4932
Browser Info        4932
Company             4932
Credit Card         4932
CC Exp Date         4932
CC Security Code    4932
CC Provider         4932
Email               4932
Job                 4932
IP Address          4932
Language            4932
Purchase Price      4932
dtype: int64

In [90]:
len(list(data.groupby('AM or PM'))[0][1])
print('Number of people who shopped during AM are ',len(AM))

Number of people who shopped during AM are  4932


In [92]:
PM=list(data.groupby('AM or PM'))[1][1]
PM

Address    Lot AM or PM  \
0     16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                      Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3                 7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
5     7502 Powell Mission Apt. 768\nTravisland, VA 3...  21 XT       PM   
...                                                 ...    ...      ...   
9992  885 Allen Mountains Apt. 230\nWallhaven, LA 16995  40 vH       PM   
9993  7555 Larson Locks Suite 229\nEllisburgh, MA 34...  72 jg       PM   
9994       6276 Rojas Hollow\nLake Louis, WY 56410-7837  93 Ex       PM   
9995      966 Castaneda Locks\nWest Juliafurt, CO 96415  92 XI       PM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   

                                           Browser Info  \
0     Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2     Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3     Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
5     Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10_8_5...   
...                                                 ...   
9992  Mozilla/5.0 (Macintosh; PPC Mac OS X 10_6_5) A...   
9993  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_8...   
9994  Opera/9.68.(X11; Linux x86_64; sl-SI) Presto/2...   
9995  Mozilla/5.0 (Windows NT 5.1) AppleWebKit/5352 ...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   

                              Company       Credit Card CC Exp Date  \
0                     Martinez-Herman  6011929061123406       02/20   
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
2          Simpson, Williams and Pham      675957666125       08/19   
3     Williams, Marshall and Buchanan  6011578504430710       02/24   
5                      Silva-Anderson    30246185196287       07/25   
...                               ...               ...         ...   
9992                        Wells Ltd  4664825258997302       10/20   
9993                   Colon and Sons    30025560104631       10/25   
9994                     Ritter-Smith  3112186784121077       01/25   
9995                    Randall-Sloan   342945015358701       03/22   
9998                         Cook Inc   180003348082930       11/17   

      CC Security Code       CC Provider                           Email  \
0                  900      JCB 16 digit               pdunlap@yahoo.com   
1                  561        Mastercard              anthony41@reed.com   
2                  699      JCB 16 digit  amymiller@morales-harrison.com   
3                  384          Discover     brent16@olson-robinson.info   
5                 7169          Discover               ynguyen@gmail.com   
...                ...               ...                             ...   
9992               431          Discover               bberry@wright.net   
9993               629           Maestro       chelseawilliams@lopez.biz   
9994              1823           Maestro              iroberts@gmail.com   
9995               838      JCB 15 digit          iscott@wade-garner.com   
9998               987  American Express         elizabethmoore@reid.net   

                                         Job       IP Address Language  \
0     Scientist, product/process development  149.146.147.205       el   
1                          Drilling engineer     15.160.41.51       fr   
2                   Customer service manager   132.207.160.22       de   
3                          Drilling engineer     30.250.74.19       es   
5                          Fish farm manager    55.96.152.147       ru   
...                                      ...              ...      ...   
9992                            Set designer    174.173.51.32       de   
9993            Designer, exhibition/display    177.46.82.128       el   
9994  

In [93]:
PM.count()

Address             5068
Lot                 5068
AM or PM            5068
Browser Info        5068
Company             5068
Credit Card         5068
CC Exp Date         5068
CC Security Code    5068
CC Provider         5068
Email               5068
Job                 5068
IP Address          5068
Language            5068
Purchase Price      5068
dtype: int64

In [94]:
len(list(data.groupby('AM or PM'))[1][1])
print('The number of people who shopped during PM are, ',len(PM))

The number of people who shopped during PM are,  5068


There is another way to do this

In [47]:
data['AM or PM'].value_counts()  #simple,aint it?!


PM    5068
AM    4932
Name: AM or PM, dtype: int64

How many people have Credit Card which expires in 2020?

In [48]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [ ]:
data['CC Exp Date'].str.contains('20')

In [56]:
data[data['CC Exp Date'].str.contains('20')]

Address    Lot AM or PM  \
0     16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
19                  125 Hall Summit\nBoothton, IL 41721  99 CU       PM   
32                     Unit 3628 Box 6778\nDPO AE 72362  39 Qm       PM   
36    9374 Skinner Common Apt. 254\nChristopherfort,...  80 Fq       PM   
38    9671 Riley Drives Apt. 746\nPort Davidtown, TN...  15 vj       AM   
...                                                 ...    ...      ...   
9883        751 Petersen Groves\nLake Rebecca, NM 55875  61 tE       PM   
9901      908 Norman Burgs\nPhillipsberg, VT 92023-9956  90 Ng       PM   
9911  44320 Gray Centers Suite 325\nJenniferfurt, OK...  19 Bf       PM   
9957             73556 Chase Pine\nBrucemouth, MP 73473  25 rU       PM   
9992  885 Allen Mountains Apt. 230\nWallhaven, LA 16995  40 vH       PM   

                                           Browser Info  \
0     Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
19    Mozilla/5.0 (compatible; MSIE 7.0; Windows NT ...   
32    Mozilla/5.0 (Windows 98; Win 9x 4.90) AppleWeb...   
36    Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
38    Mozilla/5.0 (X11; Linux i686; rv:1.9.6.20) Gec...   
...                                                 ...   
9883  Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10_8_7...   
9901  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9911  Mozilla/5.0 (X11; Linux i686; rv:1.9.7.20) Gec...   
9957  Opera/9.49.(Windows 95; en-US) Presto/2.9.189 ...   
9992  Mozilla/5.0 (Macintosh; PPC Mac OS X 10_6_5) A...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
19                 Turner-Mckinney      676343504830       02/20   
32                 Martinez-Wilson  4942281854569455       01/20   
36                     Hanna-Grant   180042289507877       09/20   
38    Bryant, Hubbard and Gonzales   210094965373094       12/20   
...                            ...               ...         ...   
9883                Perez and Sons  4939814860063025       05/20   
9901                    Hughes Inc    30148833294789       09/20   
9911                    Jones-Gray     4058074716188       12/20   
9957    Martin, Clayton and Jarvis  3337737293232646       05/20   
9992                     Wells Ltd  4664825258997302       10/20   

      CC Security Code                  CC Provider  \
0                  900                 JCB 16 digit   
19                 440                VISA 16 digit   
32                8360                 JCB 16 digit   
36                 912                VISA 16 digit   
38                 248                      Voyager   
...                ...                          ...   
9883               238                   Mastercard   
9901               483                      Voyager   
9911               921  Diners Club / Carte Blanche   
9957               798                VISA 13 digit   
9992               431                     Discover   

                             Email                                     Job  \
0                pdunlap@yahoo.com  Scientist, product/process development   
19            ruiznicole@gmail.com              Designer, interior/spatial   
32            shane21@atkinson.com             Civil Service fast streamer   
36        saundersernest@walsh.com                     Animal technologist   
38      djennings@boyd-english.org                         Music therapist   
...                            ...                                     ...   
9883            lisa25@hotmail.com                Therapeutic radiographer   
9901             james32@gmail.com                           Meteorologist   
9911  hensonlindsay@graves-fox.com                    Optician, dispensing   
9957        latashaochoa@gmail.com                Therapeutic radiographer   
9992             bberry@wright.net                            Set designer   

           IP Address L

In [57]:
data[data['CC Exp Date'].str.contains('20')].count()

Address             988
Lot                 988
AM or PM            988
Browser Info        988
Company             988
Credit Card         988
CC Exp Date         988
CC Security Code    988
CC Provider         988
Email               988
Job                 988
IP Address          988
Language            988
Purchase Price      988
dtype: int64

In [59]:
cc_20=data[data['CC Exp Date'].str.contains('20')]

In [60]:
len(cc_20)

988

In [62]:
print('The number of credit cards which expire in 2020, ', len(cc_20))

The number of credit cards which expire in 2020,  988


Now lets find the most popular email providers

In [63]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [64]:
data.Email

0                    pdunlap@yahoo.com
1                   anthony41@reed.com
2       amymiller@morales-harrison.com
3          brent16@olson-robinson.info
4          christopherwright@gmail.com
                     ...              
9995            iscott@wade-garner.com
9996                mary85@hotmail.com
9997                 tyler16@gmail.com
9998           elizabethmoore@reid.net
9999             rachelford@vaughn.com
Name: Email, Length: 10000, dtype: object

In [76]:
email=data.Email.str.split('@',expand=True)
email

0                     1
0               pdunlap             yahoo.com
1             anthony41              reed.com
2             amymiller  morales-harrison.com
3               brent16   olson-robinson.info
4     christopherwright             gmail.com
...                 ...                   ...
9995             iscott       wade-garner.com
9996             mary85           hotmail.com
9997            tyler16             gmail.com
9998     elizabethmoore              reid.net
9999         rachelford            vaughn.com

[10000 rows x 2 columns]

In [77]:
email.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       10000 non-null  object
 1   1       10000 non-null  object
dtypes: object(2)
memory usage: 156.4+ KB


In [78]:
email.iloc[:,1]

0                  yahoo.com
1                   reed.com
2       morales-harrison.com
3        olson-robinson.info
4                  gmail.com
                ...         
9995         wade-garner.com
9996             hotmail.com
9997               gmail.com
9998                reid.net
9999              vaughn.com
Name: 1, Length: 10000, dtype: object

In [75]:
email.iloc[:,1].value_counts()

hotmail.com           1638
yahoo.com             1616
gmail.com             1605
smith.com               42
williams.com            37
                      ... 
turner.info              1
hines.com                1
cuevas-porter.com        1
munoz-simmons.com        1
snyder-mercer.info       1
Name: 1, Length: 3416, dtype: int64

As is evident,the most commonly used email provider is hotmail.com

The top five email providers:


In [82]:
email.iloc[:,1].value_counts().head()

hotmail.com     1638
yahoo.com       1616
gmail.com       1605
smith.com         42
williams.com      37
Name: 1, dtype: int64

WHich professionals do the most purchasing?

In [86]:
data.Job.value_counts()

Interior and spatial designer        31
Lawyer                               30
Social researcher                    28
Research officer, political party    27
Purchasing manager                   27
                                     ..
Stage manager                         7
Editor, film/video                    6
Actuary                               6
Investment analyst                    5
Information systems manager           5
Name: Job, Length: 623, dtype: int64

Looks like the Interior designers love shopping online

In [87]:
data

Address    Lot AM or PM  \
0     16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                      Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3                 7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4     23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
...                                                 ...    ...      ...   
9995      966 Castaneda Locks\nWest Juliafurt, CO 96415  92 XI       PM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
0     Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2     Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3     Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4     Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
...                                                 ...   
9995  Mozilla/5.0 (Windows NT 5.1) AppleWebKit/5352 ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                              Company       Credit Card CC Exp Date  \
0                     Martinez-Herman  6011929061123406       02/20   
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
2          Simpson, Williams and Pham      675957666125       08/19   
3     Williams, Marshall and Buchanan  6011578504430710       02/24   
4           Brown, Watson and Andrews  6011456623207998       10/25   
...                               ...               ...         ...   
9995                    Randall-Sloan   342945015358701       03/22   
9996         Hale, Collins and Wilson   210033169205009       07/25   
9997                     Anderson Ltd  6011539787356311       05/21   
9998                         Cook Inc   180003348082930       11/17   
9999                       Greene Inc  4139972901927273       02/19   

      CC Security Code                  CC Provider  \
0                  900                 JCB 16 digit   
1                  561                   Mastercard   
2                  699                 JCB 16 digit   
3                  384                     Discover   
4                  678  Diners Club / Carte Blanche   
...                ...                          ...   
9995               838                 JCB 15 digit   
9996               207                 JCB 16 digit   
9997                 1                VISA 16 digit   
9998               987             American Express   
9999               302                 JCB 15 digit   

                               Email                                     Job  \
0                  pdunlap@yahoo.com  Scientist, product/process development   
1                 anthony41@reed.com                       Drilling engineer   
2     amymiller@morales-harrison.com                Customer service manager   
3        brent16@olson-robinson.info                       Drilling engineer   
4        christopherwright@gmail.com                             Fine artist   
...                              ...                                     ...   
9995          iscott@wade-garner.com                              Printmaker   
9996              mary85@hotmail.com                         Energy engineer   
9997               tyler16@gmail.com                      Veterinary surgeon   
9998         elizabethmoore@reid.net                Local government officer   
9999           rachelford@vaughn.com        

In [95]:
AM

Address    Lot AM or PM  \
4     23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
6        93971 Conway Causeway\nAndersonburgh, AZ 75107  96 Xt       AM   
9       3795 Dawson Extensions\nLake Tinafort, ID 88739  15 Ug       AM   
12      733 Heather Rest Apt. 670\nBoltonport, UT 78662  69 DO       AM   
14    8774 Jason Keys Suite 427\nEast Scottborough, ...  70 zH       AM   
...                                                 ...    ...      ...   
9989         5674 Cruz Trace\nNew Shelby, KY 51047-5469  12 Kf       AM   
9991              PSC 8165, Box 8498\nAPO AP 60327-0346  50 dA       AM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
4     Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
6     Mozilla/5.0 (compatible; MSIE 7.0; Windows NT ...   
9     Mozilla/5.0 (X11; Linux i686; rv:1.9.7.20) Gec...   
12    Mozilla/5.0 (Macintosh; Intel Mac OS X 10_5_3 ...   
14    Mozilla/5.0 (Windows 98; it-IT; rv:1.9.2.20) G...   
...                                                 ...   
9989  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9991  Mozilla/5.0 (compatible; MSIE 8.0; Windows NT ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                           Company       Credit Card CC Exp Date  \
4        Brown, Watson and Andrews  6011456623207998       10/25   
6                  Gibson and Sons  6011398782655569       07/24   
9     Rivera, Buchanan and Ramirez     4396283918371       01/17   
12                    Moore-Martin  5115990487067905       05/26   
14        Leach, Howe and Ferguson   869967499275071       09/22   
...                            ...               ...         ...   
9989  Vargas, Nichols and Martinez  5278049688438382       12/23   
9991                    Snyder Inc  4221582137197481       02/24   
9996      Hale, Collins and Wilson   210033169205009       07/25   
9997                  Anderson Ltd  6011539787356311       05/21   
9999                    Greene Inc  4139972901927273       02/19   

      CC Security Code                  CC Provider  \
4                  678  Diners Club / Carte Blanche   
6                  714                VISA 16 digit   
9                  931             American Express   
12                 119                VISA 16 digit   
14                 427                VISA 16 digit   
...                ...                          ...   
9989               996                VISA 16 digit   
9991               969                      Voyager   
9996               207                 JCB 16 digit   
9997                 1                VISA 16 digit   
9999               302                 JCB 15 digit   

                            Email                           Job  \
4     christopherwright@gmail.com                   Fine artist   
6              olivia04@yahoo.com                        Dancer   
9        qcoleman@hunt-huerta.com            Forensic scientist   
12              tholt@hotmail.com            Surveyor, quantity   
14               aburns@yahoo.com                 Acupuncturist   
...                           ...                           ...   
9989   walterswilliam@hotmail.com  Community development worker   
9991           kking@wise-liu.com         IT sales professional   
9996           mary85@hotmail.com               Energy engineer   
9997            tyler16@gmail.com            Veterinary surgeon   
9999        rachelford@vaughn.com        Embryologist, clinical   

           IP Address Language  Purchase Price  
4        24.140.33.94       es           77.82  
6      127.252.144.18       de           88.56  
9       

In [96]:
AM.Job.value_counts()

Social worker                                            17
Psychotherapist, dance movement                          16
Scientist, research (physical sciences)                  15
Armed forces technical officer                           15
Copywriter, advertising                                  15
                                                         ..
Information systems manager                               2
Investment analyst                                        2
Armed forces logistics/support/administrative officer     2
Trade union research officer                              1
Stage manager                                             1
Name: Job, Length: 623, dtype: int64

As is evident, Social workers do most of their shopping during the daytime

In [98]:
PM.Job.value_counts()

Social researcher                   18
Designer, interior/spatial          18
Interior and spatial designer       18
Scientist, biomedical               16
Optometrist                         16
                                    ..
Investment banker, operational       2
Therapist, occupational              2
Learning mentor                      2
Education officer, environmental     1
Editor, film/video                   1
Name: Job, Length: 623, dtype: int64

Social researchers do shopping later part of the day.

In [124]:
data.rename(columns={'Purchase Price':'Purchase_Price'},inplace=True)
data

Address    Lot AM or PM  \
0     16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                      Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3                 7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4     23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
...                                                 ...    ...      ...   
9995      966 Castaneda Locks\nWest Juliafurt, CO 96415  92 XI       PM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
0     Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2     Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3     Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4     Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
...                                                 ...   
9995  Mozilla/5.0 (Windows NT 5.1) AppleWebKit/5352 ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                              Company       Credit Card CC Exp Date  \
0                     Martinez-Herman  6011929061123406       02/20   
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
2          Simpson, Williams and Pham      675957666125       08/19   
3     Williams, Marshall and Buchanan  6011578504430710       02/24   
4           Brown, Watson and Andrews  6011456623207998       10/25   
...                               ...               ...         ...   
9995                    Randall-Sloan   342945015358701       03/22   
9996         Hale, Collins and Wilson   210033169205009       07/25   
9997                     Anderson Ltd  6011539787356311       05/21   
9998                         Cook Inc   180003348082930       11/17   
9999                       Greene Inc  4139972901927273       02/19   

      CC Security Code                  CC Provider  \
0                  900                 JCB 16 digit   
1                  561                   Mastercard   
2                  699                 JCB 16 digit   
3                  384                     Discover   
4                  678  Diners Club / Carte Blanche   
...                ...                          ...   
9995               838                 JCB 15 digit   
9996               207                 JCB 16 digit   
9997                 1                VISA 16 digit   
9998               987             American Express   
9999               302                 JCB 15 digit   

                               Email                                     Job  \
0                  pdunlap@yahoo.com  Scientist, product/process development   
1                 anthony41@reed.com                       Drilling engineer   
2     amymiller@morales-harrison.com                Customer service manager   
3        brent16@olson-robinson.info                       Drilling engineer   
4        christopherwright@gmail.com                             Fine artist   
...                              ...                                     ...   
9995          iscott@wade-garner.com                              Printmaker   
9996              mary85@hotmail.com                         Energy engineer   
9997               tyler16@gmail.com                      Veterinary surgeon   
9998         elizabethmoore@reid.net                Local government officer   
9999           rachelford@vaughn.com        

In [132]:
data.groupby('Job').Purchase_Price.mean().to_frame().sort_values(by='Purchase_Price',ascending=False)

Purchase_Price
Job                                             
Trade mark attorney                    73.802727
Translator                             70.777647
Investment analyst                     69.090000
Accountant, chartered management       69.082667
Designer, industrial/product           68.714667
...                                          ...
Theatre stage manager                  31.580000
Editor, magazine features              31.445294
Engineer, production                   27.771667
Art gallery manager                    27.381250
Marketing executive                    26.008889

[623 rows x 1 columns]

On an average